core/src/test/java/tech/pegasys/eth2signer/core/multikey/DirectoryBackedArtifactSignerProviderTest.java

```diff
@@ -0,0 +1,209 @@
+    when(signerParser.parse(pathEndsWith(filename2))).thenReturn(createArtifactSigner(PRIVATE_KEY));
+
+    final Optional<ArtifactSigner> loadedMetadataFile = signerProvider.getSigner(PUBLIC_KEY);
+
+    assertThat(loadedMetadataFile).isEmpty();
+  }
+
+  @Test
+  void signerReturnedForMetadataFileWithPrefix() throws IOException {
+    final String filename = "someprefix" + PUBLIC_KEY;
+    createFile(filename);
+    when(signerParser.parse(any())).thenReturn(artifactSigner);
+
+    final Optional<ArtifactSigner> signer = signerProvider.getSigner(PUBLIC_KEY);
+    assertThat(signer).isNotEmpty();
+    assertThat(signer.get().getIdentifier()).isEqualTo("0x" + PUBLIC_KEY);
+    verify(signerParser).parse(pathEndsWith(filename));
+  }
+
+  @Test
+  void signerIdentifiersReturnedForMetadataFile() throws IOException {
+    createFile(PUBLIC_KEY + ".yaml");
+    when(signerParser.parse(any())).thenReturn(artifactSigner);
+
+    assertThat(signerProvider.availableIdentifiers()).containsExactly("0x" + PUBLIC_KEY);
+  }
+
+  @Test
+  void signerIdentifiersReturnedForAllValidMetadataFilesInDirectory() throws IOException {
+    final String privateKey1 =
+        "0x0000000000000000000000000000000065d5d1dd92ed6b75ab662afdaeb4109948c05cffcdd299f62e58e3fb5edceb67";
+    final String publicKey1 =
+        "0x889477480fbcf2c7d32fafe50c60fc716545543a5660130e84041e6f6fce5fa471ef1e7c0cdd4380b83b8d33893e6f11";
+    createFile(publicKey1);
+    when(signerParser.parse(pathEndsWith(publicKey1)))
+        .thenReturn(createArtifactSigner(privateKey1));
+
+    final String privateKey2 =
+        "0x00000000000000000000000000000000430d79925d1bc810d2bd033178fdea98c59f29edd40e80cc7f13e92fcb05a86e";
+    final String publicKey2 =
+        "0xa7c5f1c815571d02df8ebc9b083e1a7fb4b360970cc40ebb325f3d2360dd1f9723825ea0c6fa9e398cd233ef0868a8cc";
+    createFile(publicKey2);
+    when(signerParser.parse(pathEndsWith(publicKey2)))
+        .thenReturn(createArtifactSigner(privateKey2));
+
+    final String privateKey3 =
+        "0x0000000000000000000000000000000062e4325a71315d5bb757458b560dc1957118c12466978c772c31bad86a7e3a5e";
+    final String publicKey3 =
+        "0xb458bf0b2e1d3797b2f95a0f80f715b18881f74d114c824f54452893fbe6368b32de3066e472dbeb1a43181416159606";
+    createFile(publicKey3);
+    when(signerParser.parse(pathEndsWith(publicKey3)))
+        .thenReturn(createArtifactSigner(privateKey3));
+
+    final Collection<String> identifiers = signerProvider.availableIdentifiers();
+
+    assertThat(identifiers).hasSize(3);
+    assertThat(identifiers).containsOnly(publicKey1, publicKey2, publicKey3);
+  }
+
+  private Path pathEndsWith(final String endsWith) {
+    return argThat((Path path) -> path != null && path.endsWith(endsWith + FILE_EXTENSION));
+  }
+
+  @SuppressWarnings("ResultOfMethodCallIgnored")
+  private void createFile(final String filename) throws IOException {
```

Comment:
nit: rename to "createFileInConfigsDirectory" (to be explicit about where it is being created (or else, createConfigFile)

Reply:
done